### DISCUSSION TITLE
Some feature ideas

### DISCUSSION THREAD

**Discussion by svenjacobs** [spacecowboy/Feeder#297] [Ideas]
First of all, thank you very much for Feeder! It's exactly the RSS reader I've been looking for. Free, open source and with the functionality I need 🙌🏼 

After using Feeder for a few weeks I noticed some missing functionality / minor tweaks that might improve Feeder:

- Show the number of unread articles in the title bar next to the feed name, for example `All feeds (42)`
- Implement a button (maybe next to the floating action button) that immediately scrolls to the top.
  
  This is especially important when "Mark as read on scroll" is turned on and articles are sorted by "Newest first" because when I come back to Feeder it is at the last remembered position but new articles are added to the top. So when further scrolling down those articles are marked read without that I ever noticed them.
- Give the navigation drawer ("burger menu") a slightly different background color. It currently has the same background color as the main (list) screen and there is no visual separation.

Once again thank you very much! 😃

**Comment by svenjacobs** [comment]
If you accept pull requests, I'd be happy to help when I find some time 😃

**Reply by spacecowboy** [reply]
Unread count I see no issue with.

My concern with the other two is how they fit in Material Design.

I'm not a fan of having several FABs. Neither is google: https://m3.material.io/components/floating-action-button/guidelines#2f9ec481-b30d-4dc3-89d7-feb2ca7bec4d

As for the navigation drawer. The drawer has a distinct background color which is different from the background in the list. This is visible in dark mode. Light mode could have a stronger contrast though.

**Reply by svenjacobs** [reply]
> I'm not a fan of having several FABs

It doesn't necessarily need to be a FAB. It could be a small arrow pointing up with a circular background that is placed at the bottom-center of the screen, but only when the user scrolls down. As soon as the user scrolls up, the arrow disappears.

If a UI element is not desired, maybe it could a setting "scroll to top on new articles" or something like that?

> Light mode could have a stronger contrast though

I think I would be fine with this 😃

**Reply by spacecowboy** [reply]
> If a UI element is not desired, maybe it could a setting "scroll to top on new articles" or something like that?

This is already the case. If you are scrolled to the top, you will remain at the top as new items are synced .

It is a very deliberate design decision that the scroll position does not change otherwise.

Because otherwise,  as the app syncs in the background it would be impossible to scroll the list since it would constantly throw you back to the top

**Reply by spacecowboy** [reply]
But consider this idea, just to brainstorm.

As you scroll,  the FAB hides. And a small centered button appears, like in spotify. 

![Screenshot_20240609_234745_Spotify](https://github.com/spacecowboy/Feeder/assets/223655/3c8b4a23-b271-4f35-8eb6-b1b86ac0d178)

**Reply by spacecowboy** [reply]
In spotify the button only appears when scrolling upward

**Reply by svenjacobs** [reply]
> As you scroll, the FAB hides. And a small centered button appears, like in spotify.

Yes, I had something like this in mind. Do you think it makes sense for Feeder?

**Reply by spacecowboy** [reply]
I just need to think about it a bit more. The functionality I don't mind (mark as read when scroll will be completely broken I assume). Just want to be cautious about the UX.

**Reply by svenjacobs** [reply]
Yes, of course.

**Reply by Amarok24** [reply]
Simply touching (clicking) the feed name should jump to the top. This is the most intuitive behavior, no need for a special button.

**Comment by serrq** [comment]
• Double behavior for FAB (proposal)

1. Short tap "bring me on top"
2. Long press "mark all as read"
